assets/js/googlesitekit/notifications/register-defaults.js

```diff
@@ -569,6 +570,28 @@ export const DEFAULT_NOTIFICATIONS = {
 		},
 		isDismissible: true,
 	},
+	'module-recovery-alert': {
+		Component: ModuleRecoveryAlert,
+		priority: 320,
+		areaSlug: NOTIFICATION_AREAS.BANNERS_ABOVE_NAV,
+		viewContexts: [ VIEW_CONTEXT_MAIN_DASHBOARD ],
+		isDismissible: true,
+		checkRequirements: async ( { select, resolveSelect } ) => {
+			await resolveSelect( CORE_MODULES ).getRecoverableModules();
+			const recoverableModules =
+				select( CORE_MODULES ).getRecoverableModules();
+			const recoverableModulesList = Object.keys(
+				recoverableModules || {}
+			);
+			if (
+				recoverableModules === undefined ||
+				recoverableModulesList.length === 0
```

Comment:
I'm not sure it's possible for the resolved `recoverableModules` to be anything other than an array, but also `getRecoverableModules` returns an array of slugs, not an object (like `getModules`).
```suggestion
				! recoverableModules.length
```